internal/inspect/detectors/quarto_inspect_output.go

```diff
@@ -0,0 +1,236 @@
+package detectors
+
+import (
+	"encoding/json"
+	"fmt"
+	"path/filepath"
+	"slices"
+
+	"github.com/posit-dev/publisher/internal/util"
+	"github.com/spf13/afero"
+)
+
+// Copyright (C) 2025 by Posit Software, PBC.
+
+type quartoMetadata struct {
+	Title   string `json:"title"`
+	Runtime string `json:"runtime"`
+	Server  any    `json:"server"`
+}
+
+type quartoProjectConfig struct {
+	Project struct {
+		Type       string   `json:"type"`
+		Title      string   `json:"title"`
+		PreRender  []string `json:"pre-render"`
+		PostRender []string `json:"post-render"`
+		OutputDir  string   `json:"output-dir"`
+	} `json:"project"`
+	Website struct {
+		Title string `json:"title"`
+	} `json:"website"`
+}
+
+type quartoFilesData struct {
+	Input           []string `json:"input"`
+	ConfigResources []string `json:"configResources"`
+}
+
+type quartoInspectOutput struct {
+	fs afero.Fs
```

Comment:
This was an interesting inclusion. Feels like a good opportunity for me to learn more, can you explain why this was added?

Reply:
Sure thing, there are some methods in `quartoInspectOutput` that handle system paths in different ways and methods like `util.NewAbsolutePath` require the file system provider to be passed along. Although it can be `nil` and paths will pick a default `fs`, this gives flexibility specially to instantiate a fake memory file system in tests when needed.